src/lsdb/catalog/catalog.py

```diff
@@ -248,6 +251,31 @@ def cone_search(self, ra: float, dec: float, radius: float):
         ddf_partition_map = {pixel: i for i, pixel in enumerate(pixels_in_cone)}
         return Catalog(cone_search_ddf, ddf_partition_map, filtered_hc_structure)
 
+    def polygon_search(self, polygon: SingleSphericalPolygon) -> Catalog:
+        """Perform a polygonal search to filter the catalog.
+
+        Filters to points within the polygonal region specified in ra and dec, in degrees.
+        Filters partitions in the catalog to those that have some overlap with the region.
+
+        Args:
+            polygon (SingleSphericalPolygon): The polygon to filter pixels with.
+
+        Returns:
+            A new catalog containing the points filtered to those within the
+            polygonal region, and the partitions that have some overlap with it.
+        """
+        filtered_hc_structure = self.hc_structure.filter_by_polygon(polygon)
+        pixels_in_polygon = filtered_hc_structure.get_healpix_pixels()
+        partitions = self._ddf.to_delayed()
+        partitions_in_polygon = [partitions[self._ddf_pixel_map[hp_pixel]] for hp_pixel in pixels_in_polygon]
+        filtered_partitions = [
+            polygon_filter(partition, polygon, self.hc_structure) for partition in partitions_in_polygon
```

Comment:
So I hadn't actually thought about this before, but we're assuming the partition order of the filtered_hc_structure's partition_info is correct for the hipscat_index divisions here. The filtered_hc_structure is generate from an alignment, so it is in the correct order for now, but there might be issues down the line. I made #80 for this, so I don't think you have to do anything now.

Reply:
Good point! I had not given a thought about divisions after filtering. I am not confident that the filtered catalogs keep any divisions anyway because again we create the catalog from scratch. We'll probably have to regenerate them.